edb/server/protocol/auth_ext/oauth.py

```diff
@@ -110,13 +108,15 @@ async def _handle_identity(self, user_info: data.UserInfo) -> data.Identity:
 
         return data.Identity(**result_json[0])
 
-    def _get_provider_config(self, provider_name: str) -> tuple[str, str]:
+    def _get_provider_config(
+        self, provider_name: str
+    ) -> tuple[str, str, str | None]:
         provider_client_config = util.get_config(
             self.db, "ext::auth::AuthConfig::providers", frozenset
         )
         for cfg in provider_client_config:
             if cfg.name == provider_name:
-                return (cfg.client_id, cfg.secret)
+                return (cfg.client_id, cfg.secret, cfg.additional_scope)
```

Comment:
Alternatively you can just return `cfg` here.

Reply:
Oh, that's a good idea. I think it lacks some typing that made this more awkward than hiding it behind this function, I'll take a look.

Reply:
You can declare a subtype of `typing.NamedTuple` and cast `cfg` into that.